### ISSUE TITLE
format_code_in_doc_comments removes parts of non-doc comments

### ISSUE THREAD

**Issue by vmx** [rust-lang/rustfmt#4420]
**Describe the bug**

All this happens only with `format_code_in_doc_comments=true`.

A comment gets reformatted, during this reformatting the leading parts are removed.

This is  triggered within enums and structs (perhaps other things, I haven't tried). The part that is removed has to follow the comment without a whitespace and start with a square opening bracket `[`. Everything attached to the bracket (without a whitespace) is removed. 
 
**To Reproduce**

When running this minimal example with `rustfmt --config format_code_in_doc_comments=true`

```rust
enum Minimal {
    Example,
    //[thisisremoved thatsleft
    // canbeanything
}
```

The output is:

```rust
enum Minimal {
    Example,
    /* thatsleft */
    /* canbeanything */
}
```

Instead of an enum, the top two lines could also be replaced with a struct, leading to the same buggy behavior:

```rust
struct Minimal {
    example: Example,
```

**Expected behavior**

I would expect that the code is untouched by `rustfmt`.

**Meta**

- rustfmt version: rustfmt 1.4.18-nightly (c1e9b7b 2020-06-13)
- From where did you install rustfmt?: rustup
- How do you run rustfmt: `rustfmt`, but it also happens with `cargo fmt`

**Comment by Veetaha** [comment]
I would add that the line comment is reformated to an ugly star-prefixed comment:
```rust
enum Foo {
    Bar,
    // a
    // b
}

   ↓ ↓ ↓ ↓ ↓ ↓

enum Foo {
    Bar,
    /* a
     * b */
}
```

**Comment by scampi** [comment]
The problem comes from the `Custom` comment variant in `src/formatting/comment.rs`: `//[thisisremoved` is considered as the opener of a custom comment, which is then removed in `left_trim_comment_line`:

https://github.com/rust-lang/rustfmt/blob/9b784910db3b0f1296976ff6e511ad525d0ac051/src/formatting/comment.rs#L977-L982

**Comment by ytmimi** [comment]
Going to close this as it was fixed in #5091
